core/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala

```diff
@@ -99,7 +99,11 @@ private[mesos] class MesosSubmitRequestServlet(
     // cause spark-submit script to look for files in SPARK_HOME instead.
     // We only need the ability to specify where to find spark-submit script
     // which user can user spark.executor.home or spark.home configurations.
-    val environmentVariables = request.environmentVariables.filterKeys(!_.equals("SPARK_HOME"))
+    //
+    // Do not use `filterKeys` here to avoid SI-6654, which breaks ZK persistence
```

Comment:
Should a scalastyle rule be added which prevents filterKeys to be used in future PRs ?